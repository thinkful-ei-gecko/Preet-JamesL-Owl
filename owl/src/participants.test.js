import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './participants';
import renderer from 'react-test-renderer';

//Smoke Test
describe('<Participant />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant store = {
      [
        {
          id: 3,
          name: 'James',
          avatar: '',
          inSession: true,
          onStage: false
        },
        {
          id: 4,
          name: 'Preet',
          avatar: '',
          inSession: true,
          onStage: true
        }
      ]
    } />, div);
    ReactDOM.unmountComponentAtNode(div)
  });

  it('renders UI as expected', () => {
    const tree = renderer.create(<Participant store = {
      [
        {
          id: 3,
          name: 'James',
          avatar: '',
          inSession: true,
          onStage: false
        },
        {
          id: 4,
          name: 'Preet',
          avatar: '',
          inSession: true,
          onStage: true
        }
      ]
    } />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

