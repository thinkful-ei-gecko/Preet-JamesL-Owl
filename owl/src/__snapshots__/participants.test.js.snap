// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Participant /> renders UI as expected 1`] = `
<div
  className="participant-list"
>
  <li>
    <img
      alt="avatar"
    />
    <h3>
      James
    </h3>
    <p>
      in-session
    </p>
    <p>
      
    </p>
  </li>
  <li>
    <img
      alt="avatar"
    />
    <h3>
      Preet
    </h3>
    <p>
      in-session
    </p>
    <p>
      on-stage
    </p>
  </li>
</div>
`;
